File: day48_part_3/a2.java
import java.util.HashMap;
import java.util.Map;

public class a2{
    public static void main(String[] args){
        String input = "dog cat dog bird cat dog";
        String[] words = input.split(" ");

        Map<String,Integer> wordCounts = new HashMap<>();


        for(String word : words){
            if(wordCounts.containsKey(word)){


                int currentCount = wordCounts.get(word);
                wordCounts.put(word,currentCount + 1);

            }else{
                wordCounts.put(word,1);
            }
        }

        System.out.println("Word Counts: " + wordCounts);

    }
}

File: day48_part_3/a2.py
def count_words():
    input_str = "dog cat dog bird cat dog"
    words = input_str.split(" ")
    
    word_counts = {}

    for word in words:
        if word in word_counts:
            word_counts[word] += 1
        else:
            word_counts[word] = 1

    print(word_counts)

if __name__ == "__main__":
    count_words()